### ISSUE TITLE
Create folder crashes engine

### ISSUE THREAD

**Issue by whatthesamuel** [godotengine/godot#78632]
### Godot version

v4.0.3.stable.official [5222a99f5]

### System information

Windows 10 steam

### Issue description

When I select a non-empty folder and click on the "Create Folder" button next to the project name, upon creating the project, the engine crashes. However the project gets created successfully when I restart the engine(the project shows up correctly)

### Steps to reproduce

Create new project -> select non-empty directory -> create new folder -> create project

### Minimal reproduction project

It is a problem with the engine, so there is no point of providing a minimal rep. project

**Comment by clayjohn** [comment]
I can't reproduce this on Windows 10. I did the following:
1. opened Godot 4.0.3
2. Selected "New project"
3. Selected "Browse"
4. Selected a non-empty folder
5. clicked "Create Folder"
6. clicked "Create & Edit"

Once I did so the project manager closed and the editor immediately opened in the new project I had created. 

Can you confirm that you are getting a crash when following the same steps that I just wrote out?

**Comment by whatthesamuel** [comment]
> I can't reproduce this on Windows 10. I did the following:
> 
> 1. opened Godot 4.0.3
> 2. Selected "New project"
> 3. Selected "Browse"
> 4. Selected a non-empty folder
> 5. clicked "Create Folder"
> 6. clicked "Create & Edit"
> 
> Once I did so the project manager closed and the editor immediately opened in the new project I had created.
> 
> Can you confirm that you are getting a crash when following the same steps that I just wrote out?

yes I opened this issue right after it happened. did you test it on the steam version or the standalone?

**Comment by whatthesamuel** [comment]
https://github.com/godotengine/godot/assets/31382049/be8da21f-a2a2-4f17-93ba-803532f39392

github won't allow large video files so I had to cut down the video to show the absolutely necessary part but I think this will be enough to prove my point

**Comment by clayjohn** [comment]
If you create a project without creating a new folder, does the engine start fine, or does it also freeze?

**Comment by whatthesamuel** [comment]
> If you create a project without creating a new folder, does the engine start fine, or does it also freeze?

not creating a new folder doesn't seem to help. I should have tested that before. It seems that creating a project itself is the problem then.

**Comment by clayjohn** [comment]
Can you please try creating a new project with a version of the engine from the website? 
https://godotengine.org/download/windows/

**Comment by whatthesamuel** [comment]
> Can you please try creating a new project with a version of the engine from the website? https://godotengine.org/download/windows/

It's not crashing with the standalone(website) version.

**Comment by akien-mga** [comment]
It's the same executable on Steam and on the website, so the difference may come from:
- Steam overlay (can you see if disabling it fixes it?)
- Editor settings: you can go to the Steam Godot folder, and rename the `editor_data` folder temporarily, to see if the issue is fixed with default config (if it is, then we can look at what's in your config that causes the bug)

**Comment by whatthesamuel** [comment]
> * Steam overlay (can you see if disabling it fixes it?)
This didn't work

> * Editor settings: you can go to the Steam Godot folder, and rename the `editor_data` folder temporarily, to see if the issue is fixed with default config (if it is, then we can look at what's in your config that causes the bug)
This fixed the problem. Should I provide the `editor data` folder for review or is it ok?

**Comment by akien-mga** [comment]
> This fixed the problem. Should I provide the `editor data` folder for review or is it ok?

Yes, that would be great! The whole folder would be too big because of the export templates, but most likely the relevant difference comes from the `editor_settings-4.tres`.

**Comment by whatthesamuel** [comment]
[editor_settings-4.zip](https://github.com/godotengine/godot/files/11856710/editor_settings-4.zip)

> Yes, that would be great! The whole folder would be too big because of the export templates, but most likely the relevant difference comes from the `editor_settings-4.tres`.

indeed, the file is too big. I'm only allowed to attach settings-4 file here. I'll upload the whole folder on google drive: https://drive.google.com/file/d/1xoQdWZQeRXg_wf3MjOzcRteF6MK8n5gb/view?usp=sharing
